.circleci/config.yml

```diff
@@ -32,22 +33,74 @@ jobs:
           root: ./
           paths:
             - DerivedData/Build/Products
+            - vendor/bundle
   
   Unit Tests:
     executor:
       name: ios/default
       xcode-version: "11.2.1"
     steps:
+      - git/shallow-checkout
       - ios/boot-simulator:
           xcode-version: "11.2.1"
           device: iPhone 11
       - attach_workspace:
           at: ./
+      - run:
+          name: Prepare Bundle
+          command: bundle --path vendor/bundle
       - ios/wait-for-simulator
-      - ios/xcodebuild:
-          command: test-without-building
-          arguments: -xctestrun DerivedData/Build/Products/WooCommerce_UnitTests_iphonesimulator13.2-x86_64.xctestrun -destination "platform=iOS Simulator,id=$SIMULATOR_UDID"
-      - ios/save-xcodebuild-artifacts
+      - run:
+          name: Run Unit Tests
+          command: bundle exec fastlane test_without_building xctestrun:DerivedData/Build/Products/WooCommerce_UnitTests_iphonesimulator13.2-x86_64.xctestrun destination:"platform=iOS Simulator,id=$SIMULATOR_UDID" try_count:3
+      - ios/save-xcodebuild-artifacts:
+          result-bundle-path: build/results
+  UI Tests:
+    parameters:
+      device:
+        type: string
+      post-to-slack:
+        description: Post to Slack when tests fail. SLACK_WEBHOOK ENV variable must be set.
```

Comment:
This is built into the [CircleCI/Slack orb](https://circleci.com/orbs/registry/orb/circleci/slack) we're using. You can see in the orb source that it will check for the env var and provide an error if it isn't set, so we should be covered there.